File: backend/src/services/brain/intent.detector.js
import { normalizeText } from '../../utils/text.cleaner.js';

export function detectIntent(command = '') {
  const text = normalizeText(command);

  if (/\b(abrir|abra|abre)\b/.test(text)) return { type: 'open_app', confidence: 0.92 };
  if (/\b(whatsapp|zap|mensagem|mande|enviar)\b/.test(text)) return { type: 'send_whatsapp', confidence: 0.88 };
  if (/\b(leia|ler|notificacoes|mensagens recebidas)\b/.test(text)) return { type: 'read_notifications', confidence: 0.84 };
  if (/\b(waze|maps|rota|navegar|ir para)\b/.test(text)) return { type: 'navigation', confidence: 0.82 };
  if (/\b(horas|hora atual|que horas)\b/.test(text)) return { type: 'local_time', confidence: 0.95 };
  return { type: 'ai_chat', confidence: 0.65 };
}

File: android-app/lib/core/brain/intent_detector.dart
import '../intents/megan_intent.dart';
import '../utils/text_normalizer.dart';

class IntentDetector {
  MeganIntent detect(String command) {
    final text = TextNormalizer.normalize(command);

    if (text.contains(RegExp(r'\b(abrir|abra|abre)\b'))) {
      return const MeganIntent(type: MeganIntentType.openApp, confidence: 0.92);
    }
    if (text.contains('whatsapp') || text.contains('zap') || text.contains('mensagem')) {
      return const MeganIntent(type: MeganIntentType.sendWhatsApp, confidence: 0.88);
    }
    if (text.contains('notificacao') || text.contains('notificacoes') || text.contains('leia')) {
      return const MeganIntent(type: MeganIntentType.readNotifications, confidence: 0.84);
    }
    if (text.contains('waze') || text.contains('maps') || text.contains('rota')) {
      return const MeganIntent(type: MeganIntentType.navigation, confidence: 0.82);
    }
    if (text.contains('que horas') || text.contains('hora atual')) {
      return const MeganIntent(type: MeganIntentType.localTime, confidence: 0.95);
    }
    return const MeganIntent(type: MeganIntentType.aiChat, confidence: 0.65);
  }
}
